src/lineage/entrypoint.py

```diff
@@ -180,14 +180,25 @@ def merge(repo: Repository.Repository, github_actor: str) -> Tuple[bool, List[st
     return True, conflict_file_list
 
 
-def push(repo: Repository.Repository, branch_name: str, username: str, password: str):
+def push(
+    repo: Repository.Repository, branch_name: str, username: str, password: str
+) -> bool:
     """Push changes to remote."""
-    parts: ParseResult = urlparse(repo.clone_url)
-    cred_url = parts._replace(netloc=f"{username}:{password}@{parts.netloc}").geturl()
-    logging.info("Assigning credentials for push.")
-    run([GIT, "remote", "set-url", "origin", cred_url], cwd=repo.full_name)
-    logging.info(f"Pushing {branch_name} to remote.")
-    run([GIT, "push", "--set-upstream", "origin", branch_name], cwd=repo.full_name)
+    if not repo.permissions.push:
+        logging.warning(
+            f"⚠️ ERROR! Missing 'push' permission on '{repo.owner.login}/{repo.name}'"
```

Comment:
There's a bit of a mismatch here between us logging a warning, but calling it ERROR! in the message itself.

Reply:
That's copied from:
https://github.com/cisagov/action-lineage/blob/9d4ec4d62ce3ed9e6b219bd4f9bc0e7cc6dac3f7/src/lineage/entrypoint.py#L61-L63

Reply:
I agree with @dav3r.  It should be consistent.  I'd change it anywhere else it appears as well.

Reply:
Please see 4eb817f

Reply:
Thanks @mcdonnnj 👍